libGitWrap/Tree.cpp

```diff
@@ -81,19 +81,38 @@ namespace Git
         return PrivatePtr(new Tree::Private(d->repo(), subObject));
     }
 
-    DiffList Tree::diffToTree(Result& result , Tree newTree)
+    /**
+     * @brief           Diff this tree to a newer tree.
+     *
+     * @param[in,out]   result      A Result object; see @ref GitWrapErrorHandling
+     *
+     * @param[in]       newTree     is considered as the "new tree" to compare with
+     *
+     * @return          the differences as @ref DiffList object
+     *
+     * The calling Tree object is considered as the "old tree" and @p newTree
+     * is the "new tree" to compare with. When exchanging both trees, the
+     * result will be inversed.
+     *
+     * An invalid Tree object is is considered an empty tree.
+     */
+    DiffList Tree::diffToTree(Result& result, Tree newTree)
     {
-        GW_D_CHECKED(Tree, DiffList(), result)
+        // d is allowed to be invalid
```

Comment:
I'm not really sure that I agree with d being allowed to be invalid. It _feels_ strange, even tough it has some kind of logic. Do we have any precedence of using that paradigm?